azure/services/token/client.go

```diff
@@ -0,0 +1,81 @@
+package token
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	azureautorest "github.com/Azure/go-autorest/autorest/azure"
```

Comment:
azureautorest is deprecated and we're actively working toward getting rid of it (see #2974), is it possible to not introduce new usage? Any alternative way to do what you're doing without using azureautorest libraries?

cc @mboersma

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/1a70b0108b4049f08e9918a56123233e6a586f91/azure/scope/clients.go#L45

This function returns the name instead of env, which is why i had to recalculate the env. I will update the function to return just the env. Then i don't have to figure out the cloud env all over again.

Reply:
This is still introducing a new use of go-autorest which is deprected. I think you should be able to use `azure.ARMClientOptions` (see https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/services/publicips/client.go#L37 for example usage) to avoid constructing your own ` azcore.ClientOptions` altogether

This will require reverting the last commit to go back since we no longer need to return the full env.

Reply:
done :white_check_mark: